### ISSUE TITLE
replace AER STAR data w/ Dataverse STAR data

### ISSUE THREAD

**Issue by benthestatistician** [benbhansen-stats/propertee#212]
In R/data.R:
- [ ] add descriptions of STARplus variables taken from original STAR files as found in [Dataverse archive](https://dataverse.harvard.edu/dataset.xhtml?persistentId=hdl:1902.1/10766)
- [ ] add description of STARplus `*_at_entry` variables created in /data-raw/STARdata.R.  (Which. checks if you had a classroom type value at grade k, if so filling in `cond_at_entry`, `grade_at_entry` and `school_at_entry` with that classroom type value, "`k`" and the school recorded for you at kindergarten, respectively.  If you don't have a classroom type value for grade k but you do at grade 1, it populates `cond_at_entry`, `grade_at_entry` and `school_at_entry` with your grade 1 values. And so on.)
- [ ] Add citation to STAR materials, along with [Dataverse URL](https://dataverse.harvard.edu/dataset.xhtml?persistentId=hdl:1902.1/10766) and the statment about the materials being in the public domain that I remember reading somewhere over there.
- [ ] Wherever the string "TK" is found, remove it and complete the passage as appropriate

In data-raw/STARdata.R
- [ ] program `download.file()` statement(s) as indicated in comments so as to pull in the needed files from Dataverse when necessary for building
- [ ] Do we want to leave the one experimental group observation with NAs for `*_at_entry` variables, or just remove it? (BH says leave in, unless #210 turns out to be trickier than expected.) 

(In progress on branch branch dataverse_star.)

**Comment by benthestatistician** [comment]
(Cassandra, feel free to wait until your appointment has gotten going to start work on this.)

**Comment by Cassandra291201** [comment]
Hi Ben, when you say STARTplus variable, what does that refer to? Did you mean all the variables in the STARdata dataset?

**Comment by benthestatistician** [comment]
No, I'm using "STARplus" to refer to a different data set.  Please refer to file data-raw/STARdata.R., *on the* `dataverse_star` *branch*.

**Comment by benthestatistician** [comment]
That `download.file()` line is a placeholder for the code I'd like to have there.  Ideally you'll determine a URL that those 2 files can be directly downloaded from, but that's not necessary, and I haven't verified that it's possible.  

I myself got the files by clicking "RData" in that File Access menu that you show.  If that's not working for you, try it in a different browser?  Once you do have that working, the next step is to see if you can back out a URL that works with `download.file()` from within R. If that's not possible, just document the steps to downloading the file in words.

**Comment by Cassandra291201** [comment]
I found the url under the metadata tab for each corresponding data file, the url for `STAR_Students.RData` looks like this https://dataverse.harvard.edu/api/access/datafile/666716. This should have worked with the `download.file()` function, but the url points to a tab-separated values (TSV) file instead of an .RData file. I am guessing it has something to do with file download restriction as listed in this [documentation](https://guides.dataverse.org/en/6.5/user/find-use-data.html#downloading-via-url). 
![Screenshot 2025-01-09 at 9 50 38 PM](https://github.com/user-attachments/assets/e8d83413-6152-4e70-a9dd-4489cbe63d99)

What I did then was downloading the RData files from the website and adding them to the propertee/data-raw directory. Then, I loaded them using this line `load("data-raw/STAR_Students.RData")`. It worked for me, and the rest of the lines successfully executed as well. I am wondering if adding files to the directory then loading them is a good practice? If not, please let me know and I will try to figure out if there are other links that work.

**Comment by benthestatistician** [comment]
Sounds like you're able to script up downloading of the .tsv version of the data?  If so, nice progress!  From there I'd see if that version of the data is usable.  

I expect it'll have all the same elements as the RData version, if maybe with minor differences in terms of naming conventions.  If so, it's worth our while to have spend time reconciling that version of the data with the version I had used, in the interests of winding up with a fully scriptable routine for pulling the data in.

**Comment by Cassandra291201** [comment]
@benthestatistician #218 has been merged, could we close this issue now?
